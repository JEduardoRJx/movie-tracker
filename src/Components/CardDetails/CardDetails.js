import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { toggleFav } from '../../actions';
import './CardDetails.scss';
import PropTypes from 'prop-types';

export const CardDetails = ({ movie, refreshFavs, user, toggleFav }) => {
  const {backdrop_path, id, isFavorite, overview, poster_path, release_date, title, vote_average} = movie;

  const releaseDate = new Date(release_date + 'T00:00').toString().split(' ').slice(1, 4).join(' ');

  const favStatus = isFavorite ? 'Remove Favorite' : 'Add Favorite';

  return (
    <section className="cardDetails">
      <img 
        className='movie-backdrop' 
        src={`http://image.tmdb.org/t/p/w1280${backdrop_path}`} 
        alt='movie backdrop' 
      />
      <article className='movie-details-wrapper'>
        <img 
          className='movie-poster' 
          src={`http://image.tmdb.org/t/p/w1280${poster_path}`} 
          alt='movie poster' 
        />
        <div className='movie-info-wrapper'>
          <h1 className='movie-title'>{title}</h1>
          <h2>Released: {releaseDate}</h2>
          <p>{overview}</p>
          <h2>Vote Average: {vote_average}</h2>
          <button 
            className='fav-btn' 
            type="button"
            disabled={!user}
            onClick={() => {
              toggleFav(id);
              refreshFavs(movie);
            }}
          >{favStatus}</button>
          <Link to='/'>
            <button 
              className='back-btn' 
              type='button'
            >â—€ back</button>
          </Link>
        </div>
      </article>
    </section>
  )
}

export const mapStateToProps = ({ user }) => ({
  user
})

export const mapDispatchToProps = dispatch => ({
  toggleFav: id => dispatch(toggleFav(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(CardDetails);

CardDetails.propTypes = {
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  vote_average: PropTypes.string.isRequired
}