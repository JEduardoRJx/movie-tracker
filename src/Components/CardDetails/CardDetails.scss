.cardDetails {
  height: 90vh;
  width: 100%;
  display: flex;
  background-color: #282c34;
  justify-content: center;
  align-items: center;
}

.movie-backdrop {
  height: 90%;
  width: 95%;
  object-fit: cover;
  box-shadow: 0px 0px 20px 10px rgba(0, 0, 0, 0.75)
}

.movie-details-wrapper{
  background-color: rgba(0, 0, 0, 0.5);
  height: 81%;
  width: 95%;
  z-index: 1;
  position: absolute;
  display:flex;
  justify-content: space-around;
}

.movie-poster {
  height: 100%;
}

.movie-info-wrapper {
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 50%;
}

.movie-title {
  font-size: 2em;
}

p {
  font-size: 1.25em;
}

.fav-btn-wrapper {
  font-size: 1.25em;
  border: none;
  width: 200px;
  border-radius: 5px;
  color: black;
  margin: 0px auto;
  &:hover {
    background-color: #BC8F8F;
    color: white;
  }
}

.back-btn {
  font-size: 1.25em;
  border: none;
  width: 200px;
  border-radius: 5px;
  &:hover {
    color: white;
    background-color: #BC8F8F;
  }
}