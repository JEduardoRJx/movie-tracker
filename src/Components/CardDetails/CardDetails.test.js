import React from 'react';
import { shallow } from 'enzyme';
import CardDetails from './CardDetails';

describe ('CardDetails', () => {
  let wrapper;
  it('should match the snapshot with all data passed in correctly', () => {
     wrapper = shallow(<CardDetails 
      title = 'Joker'
      releaseDate= 'Oct 04 2019'
      overview= 'During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.'
      vote_average= '8.6'
      />)

    expect(wrapper).toMatchSnapshotO();

  });
})
