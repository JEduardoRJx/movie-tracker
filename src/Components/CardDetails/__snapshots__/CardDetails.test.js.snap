// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should match the snapshot with all data passed in correctly 1`] = `
<section
  className="cardDetails"
>
  <img
    alt="movie backdrop"
    className="movie-backdrop"
    src="http://image.tmdb.org/t/p/w1280/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg"
  />
  <article
    className="movie-details-wrapper"
  >
    <img
      alt="movie poster"
      className="movie-poster"
      src="http://image.tmdb.org/t/p/w1280/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
    />
    <div
      className="movie-info-wrapper"
    >
      <h1
        className="movie-title"
      >
        Joker
      </h1>
      <h2>
        Released: 
        Oct 04 2019
      </h2>
      <p>
        During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.
      </p>
      <h2>
        Vote Average: 
        8.6
      </h2>
      <button
        className="fav-btn"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Add Favorite
      </button>
      <Link
        to="/"
      >
        <button
          className="back-btn"
          type="button"
        >
          â—€ back
        </button>
      </Link>
    </div>
  </article>
</section>
`;
